src/Models/Market.php

```diff
@@ -35,11 +36,18 @@ protected static function boot()
         parent::boot();
 
         static::creating(function (Market $market) {
-            $market->slug = $market->slug ?: Str::slug($market->city);
-            $invalidSlugs = config('locations.invalid_slugs') ?? [];
-            if (in_array($market->slug, $invalidSlugs)) {
-                $market->slug = Str::slug("{$market->slug}-{$market->state}");
+            $market->updatingSlug();
+            $other_market = Market::where('slug', $market->slug)->first();
+
+            if ($other_market) {
+                return false;
             }
+
+            $page = Page::create($market->slug, $market->title);
+
+            $market->page_id = $page->id; //@todo does not create the relations
```

Comment:
Hey @pdbreen I create the page here based on the Drew requiriments, in order to work on new feature I used the branch in the Blog package https://github.com/drewroberts/blog/pull/59/files#r606689657and create a function to satisfy `Page::create($market->slug, $market->title);` and works.

however when I set the relation `$market->page_id = $page->id;`  and then save it.... and try to use the eloquent relations, does not work.. or if I try explicit make `$page = Page::where('id',$market->page_id)->first()` does not work (always `null`) ... I use this kind of relations several times, but I dont know if its something related to the packages

Reply:
Not able to pull down the branch right now, but this line `$page->market()->associate($market);` should be removed.

Reply:
@wolfrednicolas  I had a chance to pull down the branch and take a quick look.  First, the new `Page::create(...)` method conflicts with a laravel convention of supporting a static create method.  But, its really not needed anyway.  In the Market::creating hook, use this snippet to create and link a page, but only when one hasn't been set already.
```
            if (empty($market->page_id)) {
                $page = Page::query()->create(['slug' => $market->slug, 'title' => $market->title]);
                $market->page_id = $page->id;
            }
```

For testing, you'll need to include `$this->actingAs(User::factory()->create())` before creating models to satisfy the need to pull an author id from the auth gate.

Next - the `Page` model is using a `Publishable` trait that adds a global context requiring published_at to be less than now:
```
        static::addGlobalScope('published', static function (Builder $builder) {
            $builder->where('published_at', '<', now());
        });
```

This should change to be less than or equal.  Tests run fast enough that no time may have elapsed causing this condition to fail as written resulting in the page seeming to be missing.  The `isPublished()` method in the trait should also be updated to be less than or equal as well.

Reply:
Should we update the Page factory to set `published_at` at sub 1 minute?